File: OOP/10-OOP-In-Java/Homework-OOP-In-Java/src/geometry/planeShapes/Circle.java
package geometry.planeShapes;

import geometry.baseClasses.PlaneShape;
import geometry.baseClasses.Vertice2D;
import geometry.interfaces.AreaMeasurable;
import geometry.interfaces.PerimeterMeasurable;

import java.security.InvalidParameterException;

public class Circle extends PlaneShape implements AreaMeasurable, PerimeterMeasurable {

	private double radius;
	
	public double getRadius() {
		return radius;
	}
	public void setRadius(double radius) {
		if (radius >= 0) {
			this.radius = radius;
		} else {
			throw new InvalidParameterException("Radius Cant Be negative value.");
		}		
	}
	
	public Circle(Vertice2D[] inVerticeArray, double radiusIn) {
		super(inVerticeArray);
		this.radius = radiusIn;
	}
	
	@Override
	public double getPerimeter() {
		double result = 2*(Math.PI*this.radius);
		return result;
	}
	
	@Override
	public double getArea() {
		double result = Math.PI*(Math.pow(this.radius, 2));
		return result;
	}
	
	@Override
	public String toString(){
		String result = "Type: Circle \n" + 
				super.toString() +
				"Perimeter : " + this.getPerimeter() + "\n" +
				"Area : " + this.getArea();		
		return result;
	}
}

File: OOP/08-Capsulation-Polimorfism/Homework-Encapsul-Polimorfism/Shapes/Circle.cs
using System;

namespace Shapes
{
    class Circle : IShape
    {
        private double radius;

        public double Radius
        {
            get { return this.radius; }
            set { this.radius = value; }
        }

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double CalculateArea()
        {
            double rPow = Math.Pow(this.Radius, 2);
            double area = Math.PI * rPow;
            return area;
        }

        public double CalculatePerimeter()
        {
            double r = this.Radius;
            double perimeter = (r*r)*Math.PI;
            return perimeter;
        }
    }
}
